integration-tests/src/test_loop/utils/loop_action.rs

```diff
@@ -0,0 +1,42 @@
+use std::cell::Cell;
+use std::rc::Rc;
+
+use near_async::test_loop::data::TestLoopData;
+use near_primitives::types::AccountId;
+
+use crate::test_loop::env::TestData;
+
+/// Signature of functions callable from inside the inner loop of a testloop test.
+pub(crate) type LoopActionFn = Box<dyn Fn(&[TestData], &mut TestLoopData, AccountId)>;
+
+/// Wraps a callable to be used inside the inner loop of a testloop test.
+/// Stores a flag denoting whether the callable can be considered successful
+/// if wouldn't be called again.
```

Comment:
What do you mean here? 

Are there expectations for resetting succeeded to false before each iteration? 

It seems like the loop action has two failure modes, can you clarify that as well please? 
* assert failure during the action execution
* succeeded equal to false at the end of the test